### ISSUE TITLE
The specified key does not exist after 2.3.9

### ISSUE THREAD

**Issue by maxbax** [aws-amplify/aws-sdk-android#321]
After #310 fixed, the version 2.4.5 resolve the space problem but not the general problem. If I write a special char into key string the upload doesn't work well.
The problem is not only for space, but for all special chars.
For example you can use "😂" as key
Usually in the key I write the file name so every characters is admitted. With the 2.3.9 version I can do this

Thanks

**Comment by defaultbr** [comment]
sorry the reference...


+1 for me it occur too

WORD: "Saída" (accent at i) the file is never uploaded to S3 (cant find it)

**Comment by kwong93** [comment]
I can also confirm, I uploaded file with the key 2017-08-01@02-47-38-M-10666-e713e147-a58a-4327-920a-b3ea006c15de.jpg and it converted the '@' symbol to %40.

**Comment by driskell** [comment]
Possibly similar - attempting to download an object containing an @ (“at”) symbol returns AccessDenied (because it cannot find it)

**Comment by kvasukib** [comment]
Thank you for reporting to us. We acknowledge that this bug has been introduced after 2.4.1 and we are working on to fix it. We will update this thread once we fix the issue. Thank you for the patience.

**Comment by minbi** [comment]
We don't have an eta on the fix, but I have marked the issue so the bot will not close.

**Comment by ioannist** [comment]
There is a similar problem with the ":" character. Using the Java SDK, I obtain the key of an S3 Put event from:

`String textFileKey = event.getS3().getObject().getKey();`

If I try to Get this file, AWS will say it does not exist. To fix it, you must replace the "%3A" encoded characters of the key with ":", a rather not elegant solution.

**Comment by kvasukib** [comment]
Sorry for the delayed reply. This issue has been fixed in 2.7.4. Please upgrade and check and let us know if you have any issues.

**Comment by AghoPwn** [comment]
I still have issues with 2.74, i'm sticking with 2.2.
I have spaces in my keys names.
With 2.2, i don't encode them and they are downloaded without issues.
With newer versions, i tried to Urlencode the key, replace space character with %20, or let the spaces unencoded, and i still can't download the file. What is the correct way to download files with space in the key name ?

Thanks,

**Comment by kvasukib** [comment]
This is a duplicate of issue #526. We are tracking this issue in #526. We will capture all the details there.
